### PR TITLE
Add an option to create the UEFI flatcar boot entry

### PR DESCRIPTION
**Pull request by jbguerraz**
# flatcar-install: Add an option to create the UEFI flatcar boot entry

## How to use

flatcar-install -u ...

## Testing done

installed a node

- [ ] flatcar-install: Add an option to create the UEFI flatcar boot entry

### CODE REVIEW

File: bin/flatcar-install [hunk 1]
```diff
@@ -669,6 +671,15 @@ function write_ignition() if [[ -n "${IGNITION}" ]]; then
     cp "${IGNITION}" "${WORKDIR}/oemfs/config.ign"
 fi
 
+function create_uefi() {
+    ensure_tool "efibootmgr"
+    local EFI_APP="bootx64.efi"
+    if [[ "${BOARD}" = "arm64-usr" ]]; then
+        EFI_APP="bootaa64.efi"
+    fi
+    efibootmgr -c -d ${DEVICE} -l "\efi\boot\${EFI_APP}" -L "Flatcar Container Linux ${CHANNEL_ID}"
```
Comment (pothos) [inline bin/flatcar-install]:
```suggestion
    efibootmgr -c -d "${DEVICE}" -l "\efi\boot\${EFI_APP}" -L "Flatcar Container Linux ${CHANNEL_ID}"
```
Maybe not needed but still good to quote for consistency

File: bin/flatcar-install [hunk 2]
```diff
@@ -416,6 +417,7 @@ do
         k) KEYFILE="$OPTARG" ;;
         f) IMAGE_FILE="$OPTARG" ;;
         n) COPY_NET=1;;
+        u) CREATE_UEFI=1;;
```
Comment (pothos) [inline bin/flatcar-install:421]: Please define it in the block around line 398: `CREATE_UEFI=`

**Comment by pothos** [comment]
Thanks for the PR!

Regarding:
> [ ] flatcar-install: Add an option to create the UEFI flatcar boot entry

The changelog entry will be added in the `changelog/changes/` folder in the coreos-overlay repo when bumping the commit ID in `coreos-base/coreos-init/coreos-init-9999.ebuild`. Can you create this PR, too? In the beginning you can use your branch's commit ID but the final one will be the ID of the merge commit when this PR here gets merged.

File: bin/flatcar-install (hunk 2 above)
Comment (pothos) [reply bin/flatcar-install:421]: Ah, and please add it to `print_settings`, too (Edit: Or not? COPY_NET isn't there either…)

Comment (jbguerraz) [reply bin/flatcar-install:421]: np. but for my information, I wonder why, what differs from `INSTALL_TO_SMALLEST=1` or `COPY_NET=1`, what's the reasoning ?

Comment (pothos) [reply bin/flatcar-install:421]: Yeah, then let's not add it to `print_settings`

Comment (pothos) [reply bin/flatcar-install:421]: Oh, the other variables are getting used undefined, I think it's not the best to write a shell script that way, so I suggested to define it first. If there are other variables not following this, no need to care about them now but eventually I hope this script would use `set -u`.

File: bin/flatcar-install [hunk 3]
```diff
@@ -105,6 +105,7 @@ Options:
     -k KEYFILE  Override default GPG key for verifying image signature.
     -f IMAGE    Install unverified local image file to disk instead of fetching.
     -n          Copy generated network units to the root partition.
+    -u          Create UEFI boot entry
```
Comment (pothos) [inline bin/flatcar-install:108]:
```suggestion
    -u          Create UEFI boot entry (requires efibootmgr, not available on Flatcar)
```
Maybe we need to document that it won't work when used on Flatcar (e.g., via PXE)? Or do you plan to add the efibootmgr binary to Flatcar?

Comment (jbguerraz) [reply bin/flatcar-install:108]: Adding it there https://github.com/flatcar-linux/coreos-overlay/pull/1955

**Comment by jbguerraz** [comment]
> Thanks for the PR!
> 
> Regarding:
> 
> > [ ] flatcar-install: Add an option to create the UEFI flatcar boot entry
> 
> The changelog entry will be added in the `changelog/changes/` folder in the coreos-overlay repo when bumping the commit ID in `coreos-base/coreos-init/coreos-init-9999.ebuild`. Can you create this PR, too? In the beginning you can use your branch's commit ID but the final one will be the ID of the merge commit when this PR here gets merged.

Bump PR https://github.com/flatcar-linux/coreos-overlay/pull/1956

**Review by pothos — APPROVED** [review]

**Review by jepio — APPROVED** [review]

**Comment by pothos** [comment]
Tested it with qemu and the entry got created as expected and after the reboot the second disk got used instead of the primary one

**Pull request merged**